src/plots/layout_attributes.js

```diff
@@ -228,6 +228,20 @@ module.exports = {
                 'between the plotting area and the axis lines'
             ].join(' ')
         },
+        minreducedwidth: {
+            valType: 'number',
+            min: 2,
+            dflt: 64,
+            editType: 'plot',
+            description: 'Minimum width of the plot with automargin applied (in px)'
+        },
+        minreducedheight: {
+            valType: 'number',
+            min: 2,
+            dflt: 64,
+            editType: 'plot',
+            description: 'Minimum height of the plot with automargin applied (in px)'
+        },
```

Comment:
@alexcjohnson Is there a better place to put these new attributes in?
It looks like these are related to `automargin` not `margin` itself.

Reply:
@archmoj and I discussed this a bit earlier today, and I think the conclusion we came to was: because `margin.autoexpand` controls whether or not these attributes are even relevant, (1) they should stay here, so they're in the same container as `autoexpand`, and (2) if `autoexpand` is `false` we shouldn't even coerce these two new ones. (The general rule is, if an attribute doesn't do anything because of the value of some other attribute(s), it should not be coerced)

The only question is whether it's really true that these attributes are irrelevant when `autoexpand=false`. According to its description this would disable automargin from "Legends, colorbars, updatemenus, sliders, axis rangeselector and rangeslider" but what if `xaxis.automargin=true`? Maybe then the margins can still be pushed even if `margin.autoexpand=false`? That needs to be tested before we make coercion conditional on `margin.autoexpand` - but even if that's the case I think it makes sense to leave these attributes where they are.

Reply:
Addressed in https://github.com/plotly/plotly.js/pull/6307/commits/93b05d3bd80b824a5dcb95e2176a4e5c4f10d5d8.

Reply:
yes, @archmoj and I looked into the source code and decided that these attributes would be best placed at the top level of layout parameters as they could be related to `autoexpand` as well (and I think logically make sense alongside `height` and `width`).